cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,303 @@
+                url,
+                headers=headers,
+                auth=(nexpose_user, nexpose_password),
+                verify=nexpose_verify_cert,
+                params=params,
+                timeout=nexpose_timeout,
+            )
+            resp.raise_for_status()
+
+            data = resp.json()
+
+            total_resources = data["page"]["totalResources"]
+            if total_resources < limit:
+                limit = total_resources
+
+        except requests.HTTPError as exception:
+            logger.warning("GetAssets0: %s", resp.content)
+            logger.exception("GetAssets0: %s", exception)
+            return result_array
+
+        try:
+            if "resources" not in data:
+                logger.warning("data without resources: %s", data)
+
+        except Exception as exception:
+            logger.warning("GetAssets1: %s", resp.content)
+            logger.exception("GetAssets1 - Incomplete data: %s", exception)
+            return result_array
+
+        try:
+            # logger.debug("GetAssets count %s", len(data['resources']))
+
+            result_array = result_array + data["resources"]
+
+            page += 1
+            count += size_interval
+
+        except Exception as exc:
+            logger.warning("GetAssets3: %s", resp.content)
+            logger.exception("GetAssets3 - Incomplete data: %s", exc)
+            return result_array
+
+    # logger.debug("result_array %s", (result_array))
+
+    logger.info("Flattening data")
+    df_rapid7_tmp = pandas.json_normalize(result_array, sep="_", max_level=3)
+    df_rapid7_tmp["subscription_id"] = df_rapid7_tmp["configurations"].apply(
+        extract_rapid7_configurations_subscriptionid,
+    )
+    df_rapid7_tmp["resource_group"] = df_rapid7_tmp["configurations"].apply(
+        extract_rapid7_configurations_resourcegroup,
+    )
+    df_rapid7_tmp["resource_id"] = df_rapid7_tmp["configurations"].apply(
+        extract_rapid7_configurations_resourceid,
+    )
+    df_rapid7_tmp["instance_id"] = df_rapid7_tmp["configurations"].apply(
+        extract_rapid7_configurations_instanceid,
+    )
+    df_rapid7_tmp["tool_last_seen"] = df_rapid7_tmp["history"].apply(
+        extract_rapid7_history_lastseen,
+    )
+    df_rapid7_tmp["tool_first_seen"] = df_rapid7_tmp["history"].apply(
+        extract_rapid7_history_firstseen,
+    )
+    logger.debug("Example df_rapid7_tmp: %s", df_rapid7_tmp.head(1))
+    flatten_data = json.loads(df_rapid7_tmp.to_json(orient="records"))
+    logger.debug("Example: %s", flatten_data[0])
+
+    return flatten_data
+
+
+# pylint: disable=unused-argument
+def Rapid7VulnerabilitiesByHost(
+    authorization: tuple[str, str, str, bool],
+    machine_id,
+) -> array:
+    """
+    Get Rapid7 vulnerabilities
+
+    https://help.rapid7.com/insightvm/en-us/api/index.html#operation/getVulnerabilities
+    Vulnerabilities per host are returned with hosts
+    """
+
+    # TBD
+
+    return ["TBD"]
```

Comment:
Still TBD?

Reply:
Function removed as the host vulnerabilities part is not on my priorities for now.